docs/about-core-concepts/index.rst

```diff
@@ -31,6 +30,46 @@ All software in the Open Data Cube project family is released under the `Apache
 .. _datacube-core: https://github.com/opendatacube/datacube-core
 
 
+The Open Data Cube Software Ecosystem
+=====================================
+
+The Open Data Cube is a software ecosystem consisting of a number of Python packages.
+
+The most important packages in the ecosystem are described in the `ODC Software Packages`_
+section.
+
+Two packages in particular have overlapping functionality and are worthy of early attention:
+
+.. _`ODC Software Packages`: extensions.html
+
+Datacube-core or odc-stac?
+--------------------------
+
+`datacube-core`_ provides many powerful tools and services, however the needs of many new users
+can be met more easily with `odc-stac`_.
+
+Both `odc-stac`_ and `datacube-core`_ allow searching collections of earth observation data and loading
+it into xarray data structures.  The main differences are in the metadata model that each use
+(STAC for `odc-stac`_ and `eo3`_ for `datacube-core`),
+and in performing search and discovery via STAC-API endpoints (in the case of `odc-stac`_)
+or a relational database (in the case of `datacube-core`_).
+
+If you have access to an ODC database that somebody else
+maintains, you probably already have access to an environment with datacube-core installed and configured.
+If you want to create and maintain your own ODC database, you will need to install `datacube-core`_.  Most
+other use case scenarios should look at `odc-stac`_ first.
+
+This diagram illustrates the relationships between, and highlights the differences between `odc-stac`_ and
+`datacube-core`_, as of datacube 1.9:
+
+.. figure:: ../diagrams/odc1.9-arch.png
```

Comment:
I think this may be too much detail to help a user decide whether to use datacube-core or odc-stac. I would suggest we make either a flow chart or a comparison table that guides the user to the product they want to use. I can have a go at this.